File: Python/#1507_reformatDate.py
class Solution:
    def reformatDate(self, date: str) -> str:
        temp=date.split(" ")
        res=temp[2]+"-"
        month=["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
        m=0
        for i in range(len(month)):
            if temp[1]==month[i]:
                m=i+1
                break
        if m<10:
            res+="0"+str(m)+"-"
        else:
            res+=str(m)+"-"
        d=temp[0]
        dd=d[:-2]
        if int(dd)<10:
            res+="0"+dd
        else:
            res+=dd
        return res
                
        

File: Java/#1507_reformatDate.java
class Solution {
    public String reformatDate(String date) {
        String res="";
        String[] temp=date.split(" ");
        res+=temp[2]+"-";
        String[] month={"Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"};
        int m=0;
        for(int i=0;i<month.length;i++){
            if(temp[1].equals(month[i])){
                m=i+1;
                break;
            }
        }
        if(m<10){
            res+="0"+m+"-";
        }else{
            res+=m+"-";
        }
        String d=temp[0].substring(0,temp[0].length()-2);
        int n = Integer.parseInt(d);
        if(n<10){
            res+="0"+d;
        }else{
            res+=d;
        }
        return res;
    }
}
